tests/tenderly-simulation-new.ts

```diff
@@ -0,0 +1,534 @@
+    return ethers.utils.keccak256(
+      ethers.utils.concat([balanceOfSlot, ethers.utils.hexZeroPad(owner, 32)]),
+    );
+  }
+
+  /**
+   *
+   * @param allowanceSlot storage slot of `allowance` mapping
+   * @param owner account's address
+   * @param spender spender's address
+   * @param isVyper `true` if contract is written in Vyper
+   */
+  calculateAddressAllowanceSlot(
+    allowanceSlot: string,
+    owner: string,
+    spender: string,
+    isVyper = false,
+  ) {
+    return isVyper
+      ? this.calculateVyperAddressAllowanceSlot(allowanceSlot, owner, spender)
+      : this.calculateSolidityAddressAllowanceSlot(
+          allowanceSlot,
+          owner,
+          spender,
+        );
+  }
+
+  /**
+   *
+   * @param allowanceSlot storage slot of `allowance` mapping
+   * @param owner account's address
+   * @param spender spender's address
+   */
+  calculateSolidityAddressAllowanceSlot(
+    allowanceSlot: string,
+    owner: string,
+    spender: string,
+  ) {
+    const slotHash = ethers.utils.keccak256(
+      ethers.utils.concat([
+        ethers.utils.hexZeroPad(owner, 32),
+        ethers.utils.hexZeroPad(allowanceSlot, 32),
+      ]),
+    );
+
+    return ethers.utils.keccak256(
+      ethers.utils.concat([ethers.utils.hexZeroPad(spender, 32), slotHash]),
+    );
+  }
+
+  /**
+   *
+   * @param allowanceSlot storage slot of `allowance` mapping
+   * @param owner account's address
+   * @param spender spender's address
+   */
+  calculateVyperAddressAllowanceSlot(
+    allowanceSlot: string,
+    owner: string,
+    spender: string,
+  ) {
+    const slotHash = ethers.utils.keccak256(
+      ethers.utils.concat([
+        ethers.utils.hexZeroPad(allowanceSlot, 32),
+        ethers.utils.hexZeroPad(owner, 32),
+      ]),
+    );
+
+    return ethers.utils.keccak256(
+      ethers.utils.concat([slotHash, ethers.utils.hexZeroPad(spender, 32)]),
+    );
+  }
+
+  buildBalanceOfSimulationRequest(
+    chainId: number,
+    token: string,
+    owner: string,
+  ): SimulateTransactionRequest {
+    const iface = new ethers.utils.Interface([
+      'function balanceOf(address owner) view returns (uint)',
+    ]);
+
+    return {
+      from: ethers.constants.AddressZero,
+      to: token,
+      data: iface.encodeFunctionData('balanceOf', [owner]),
+      chainId,
+    };
+  }
+
+  buildAllowanceSimulationRequest(
+    chainId: number,
+    token: string,
+    owner: string,
+    spender: string,
+  ): SimulateTransactionRequest {
+    const iface = new ethers.utils.Interface([
+      'function allowance(address owner, address spender) view returns (uint)',
+    ]);
+
+    return {
+      from: ethers.constants.AddressZero,
+      to: token,
+      data: iface.encodeFunctionData('allowance', [owner, spender]),
+      chainId,
+    };
+  }
+
+  async findTokenBalanceOfSlot(
+    chainId: number,
+    token: string,
+  ): Promise<{ slot: string; isVyper?: boolean }> {
+    const account = ethers.constants.AddressZero;
```

Comment:
I think we can use any non-zero address to make we skip all contract's side logic related to handling ZERO addresses (hard to believe that any contract can handle it separately for `balanceOf` or `allowance` call, but still why not to prevent this)

Reply:
fixed in https://github.com/paraswap/paraswap-dex-lib/pull/886/commits/ed5740ba2baa5a3f78f5d516afa2e8203b22c493, using recognisable names now